test/utils.test.ts

```diff
@@ -117,10 +123,238 @@ describe('utils', function () {
         .onSecondCall()
         .resolves(availableRelayHub);
       const {
-        relayInfo: { url: actualUrl },
+        managerData: { url: actualUrl },
       } = (await selectNextRelay(httpClientStub)) as RelayInfo;
 
       expect(actualUrl).to.equal(expectedUrl);
     });
   });
+
+
+  describe('validateRelayResponse', function () {
+    let deployRequest: DeployTransactionRequest;
+    let relayWorkerAddress: string;
+    let relayHubAddress: string;
+
+    beforeEach(function () {
+      relayWorkerAddress = createRandomAddress();
+      relayHubAddress = FAKE_ENVELOPING_CONFIG.relayHubAddress;
+
+      deployRequest = {
+        metadata: {
+          relayHubAddress,
+          relayMaxNonce: 6,
+          signature: '',
+        },
+        relayRequest: {
+          relayData: {
+            callForwarder: '',
+            callVerifier: '',
+            feesReceiver: '',
+            gasPrice: '',
+          },
+          request: {
+            from: '',
+            to: '',
+            index: 0,
+            nonce: 0,
+            recoverer: createRandomAddress(),
+            relayHub: '',
+            tokenAmount: 0,
+            tokenContract: '',
+            tokenGas: 0,
+            value: 0,
+            data: '',
+          },
+        },
+      };
+
+      const encodeFunctionDataStub = sinon.stub();
+
+      encodeFunctionDataStub.withArgs('relayCall', sinon.match.any).returns('Dummy Relay Data')
+      encodeFunctionDataStub.withArgs('deployCall', sinon.match.any).returns('Dummy Deploy Data');
+      const relayHubStub = {
+        interface: {
+          encodeFunctionData: encodeFunctionDataStub
+        }
+      } as unknown as RelayHub;
+      sinon.stub(RelayHub__factory, 'connect').returns(relayHubStub);
+    });
+
+    afterEach(function () {
+      sinon.restore();
+    });
+
+    it('Should perform checks on deploy transaction and not throw errors', function () {
+      // FIXME: this test sounds meaningless.
```

Comment:
Indeed I don't find any difference between this one and the next one at line 209

Reply:
One is for deployment and the other one is regular transaction, should I keep both.

Reply:
No, I wouldn't. I know that logically they're different scenarios, but if the request is exactly the same, so I wouldn't replicate it (we could add a comment to specify it).

Reply:
updated